listenbrainz/webserver/views/profile.py

```diff
@@ -258,3 +261,23 @@ def connect_spotify_callback():
         flash.warn('Unable to authenticate with Spotify (error {})'.format(e.args[0]))
 
     return redirect(url_for('profile.connect_spotify'))
+
+
+@profile_bp.route('/refresh-spotify-token', methods=['GET'])
+@crossdomain()
+@api_login_required
+def refresh_spotify_token():
+    spotify_user = spotify.get_user(current_user.id)
+    if not spotify_user:
+        raise APINotFound("User has not authenticated to Spotify")
```

Comment:
I don't know if 404 is a good error here. The page still exists, it just can't be run. 400 doesn't seem right either:
> the server cannot or will not process the request due to something that is perceived to be a client error

Maybe 409?